app/src/main/java/org/mozilla/fenix/perf/ProfilerReusableComposable.kt

```diff
@@ -40,35 +44,39 @@ fun ProfilerDialogueCard(content: @Composable () -> Unit) {
 }
 
 /**
- * Top level radio button for the profiler dialogue
+ * Top level radio button for the profiler dialogue.
+ *
+ * @param text The main text to be displayed.
+ * @param subText The subtext to be displayed.
+ * @param selected [Boolean] that controls if radio button is selectable.
+ * @param onClick Callback to be invoked when radio button is clicked.
  */
 @Composable
 fun ProfilerLabeledRadioButton(
     text: String,
     subText: String,
-    state: MutableState<String>,
+    selected: Boolean = false,
+    onClick: () -> Unit,
 ) {
     Row {
         RadioButton(
-            selected = state.value == text,
-            onClick = { state.value = text },
+            selected = selected,
+            onClick = onClick,
             enabled = true,
         )
         Column {
             Text(
                 text = text,
                 modifier = Modifier
                     .padding(start = 8.dp)
-                    .clickable {
-                        state.value = text
-                    },
+                    .clickable { onClick(); },
             )
             Text(
                 text = subText,
                 fontWeight = FontWeight.ExtraLight,
                 modifier = Modifier
                     .padding(start = 8.dp)
-                    .clickable { state.value = text },
+                    .clickable { onClick(); },
```

Comment:
This implementation seems off to me. Each item shouldn't have the `onClick` lambda. If we're going to refactor this screen (it's only used for profiling so not for the userbase at large), let's find a way to reduce some of the redundancy. Maybe something along the lines of your preview for `RadioButton`

Reply:
I don't fully know what this visually looks like so take this comment with the assumption that I can be completely wrong. Since every item in inside the containing Row is clickable, then I would imagine we should just make the row clickable similar to a list item.

Reply:
that's a good point, I'll make the entire row clickable.